src/components/Api.js

```diff
@@ -0,0 +1,147 @@
+export class Api {
+  constructor({ baseUrl, headers }) {
+    this._url = baseUrl;
+    this._authorization = headers.authorization;
+  }
+
+  getInitialCards() {
+    return fetch(`${this._url}/cards`, {
+      headers: {
+        authorization: this._authorization,
+        "Content-Type": "application/json",
+      },
+    })
+      .then((res) =>
+        res.ok
+          ? res.json()
+          : Promise.reject("An error has occurred", res.status)
+      )
+      .then((result) => {
+        return result;
+      })
+      .catch((err) => {
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very end of any request and this end is in `index.js`   after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.

Please, remove it from `Api` and add it in `index.js`.